public/myScript/orders.js

```diff
@@ -0,0 +1,158 @@
+
+
+var dynamicTable = (function() {
+
+    var _tableId, _table,
+        _fields, _headers,
+        _defaultText;
+
+    /** Builds the row with columns from the specified names.
+     *  If the item parameter is specified, the memebers of the names array will be used as property names of the item; otherwise they will be directly parsed as text.
+     */
+
+    function format(item){
+            var tagcolor = '<tr>';
+            switch (item){
+                case 4:tagcolor = '<tr bgcolor = " #00FF7F">';break;
+                case 2:
+                case 3:tagcolor = '<tr bgcolor = "#FFFF00">';break;
+                case 0:
+                case 1:tagcolor = '<tr bgcolor = "#FF0000">';break;
+                default: tagcolor = '<tr>'; break
+            }
+            return tagcolor;
+        }
+
+    function translate(item){
+        var tagcolor = '<tr>';
+        switch (item){
+            case 4:item = 'Новый';break;
+            case 5:item = 'Отправлен водителю';break;
+            case 6:item = 'Принят водителем';break;
+            case 3:item = 'Отклонен 1 водителем';break;
+            case 2:item = 'Нет ответа от 1 водителя';break;
+            case 1:item = 'Отклонен 2 водителем';break;
+            case 0:item = 'Нет ответа от 2 водителя';break;
+            case 7:item = 'Водитель на месте';break;
+            case 8:item = 'Клиент в машине';break;
+            case 9:item = 'Выполнен';break;
+            case 10:item = 'Отменен';break;
+        }
+        return item;
+    }
+
+    function _buildRowColumns(names, item) {
+        var rowSt = '<tr>';
+        var row ='';
+        if (names && names.length > 0)
+        {
+            $.each(names, function(index, name) {
+                var c = item ? item[name+''] : name;
+                if (name == "status"){
+                    rowSt = format(c);
+                    c=translate(c);
+                }
+                if (c == null) c='';
```

Comment:
Write code in such way, that these condition check will not be necessary. You have full control over variable `c`, make sure that its value defaults to empty string and not null.